scipy/optimize/optimize.py

```diff
@@ -786,8 +784,13 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
     one2np1 = list(range(1, N + 1))
     fsim = np.empty((N + 1,), float)
 
+    fcalls, func = _wrap_scalar_function_with_validation(func, args, maxfun)
+
     for k in range(N + 1):
-        fsim[k] = func(sim[k])
```

Comment:
This needs to be:

```
try:
    for k in range(N + 1):
        fsim[k] = func(sim[k])
except _MaxFuncCallError:
    pass
finally:
    ind = np.argsort(fsim)
    sim = np.take(sim, ind, 0)
    fsim = np.take(fsim, ind, 0)
```

This is because the `OptimizeResult.x` and `OptimizeResult.fun` are taken from `sim[0]` and `fsim[0]` so these need to be updated even if this loop doesn't run fully.